### PR TITLE
Generalizing surgery to /mob/living.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Had to implement a couple of helpers with a lot of find and replace, hence the big diff.
- `get_species()` is now `get_species_name()` with `get_species()` returning the species datum.
- New helper for internal organ tag stuff, `get_internal_organ(BP_FOO)`.
- New helper for internal organs, `get_internal_organs()`.
- Surgery now expects input mob of /mob/living at base and will fail out when appropriate for human surgeries.

### CODE REVIEW

File: code/modules/surgery/limb_reattach.dm [hunk 1]
```diff
@@ -50,26 +54,27 @@
...
 	return ..()
 
-/decl/surgery_step/limb/attach/can_use(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
-	if(..())
+/decl/surgery_step/limb/attach/can_use(mob/living/user, mob/living/target, target_zone, obj/item/tool)
+	if(ishuman(target) && ..())
+		var/mob/living/carbon/human/H = target
 		var/obj/item/organ/external/E = tool
-		var/obj/item/organ/external/P = target.organs_by_name[E.parent_organ]
+		var/obj/item/organ/external/P = H.organs_by_name[E.parent_organ]
```
Comment (comma) [inline code/modules/surgery/limb_reattach.dm]: get_organ?

File: code/modules/surgery/limb_reattach.dm [hunk 2]
```diff
@@ -30,11 +31,14 @@
...
-/decl/surgery_step/limb/attach/pre_surgery_step(mob/living/user, mob/living/carbon/human/target, target_zone, obj/item/tool)
+/decl/surgery_step/limb/attach/pre_surgery_step(mob/living/user, mob/living/target, target_zone, obj/item/tool)
+	if(!ishuman(target))
+		return FALSE
 	. = FALSE
+	var/mob/living/carbon/human/H = target
 	var/obj/item/organ/external/E = tool
-	var/obj/item/organ/external/P = target.organs_by_name[E.parent_organ]
-	var/obj/item/organ/external/T = target.organs_by_name[E.organ_tag]
+	var/obj/item/organ/external/P = H.organs_by_name[E.parent_organ]
```
Comment (comma) [inline code/modules/surgery/limb_reattach.dm]: here and line below too

**Comment by comma** [comment]
```
mods/mobs/borers/mob/borer/borer_powers.dm:66:error: : bad assignment

	H.internal_organs_by_name[BP_BRAIN] = B
```

**Review by comma — DISMISSED** [review]

**Comment by MistakeNot4892** [comment]
Tested:
- slime surgery
- incision, bone cutting, bone repair
- separating organ, removing organ, replacing organ, reconnecting organ

The other surgeries are more or less all just arg type changes, pretty sure this is good to go.

**Review by comma — DISMISSED** [review]

**Review by comma — APPROVED** [review]

**Pull request merged**